opta/gcp/flyte.yaml

```diff
@@ -0,0 +1,143 @@
+    variables:
+      google_project_id: <project_id>
+name: service-flyte
+modules:
+  - name: postgres
+    type: gcp-postgres
+  - name: gcs
+    type: gcp-gcs
+    bucket_name: "{parent_name}-{layer_name}"
+  - name: adminflyterole
+    type: gcp-service-account
+    explicit_name: gsa-flyteadmin
+    allowed_k8s_services:
+      - namespace: flyte
+        service_account_name: flyteadmin
+    links:
+      - gcs: [ "write" ]
+  - name: datacatalogrole
+    type: gcp-service-account
+    explicit_name: gsa-datacatalog
+    allowed_k8s_services:
+      - namespace: flyte
+        service_account_name: datacatalog
+    links:
+      - gcs: [ "write" ]
+  - name: flytepropellerrole
+    type: gcp-service-account
+    explicit_name: gsa-flytepropeller
+    allowed_k8s_services:
+      - namespace: flyte
+        service_account_name: flytepropeller
+    links:
+      - gcs: [ "write" ]
+  - name: flyteproductionrole
+    type: gcp-service-account
+    explicit_name: gsa-production
+    allowed_k8s_services:
+      - namespace: production
+        service_account_name: default
+    links:
+      - gcs: [ "write" ]
+  - name: flytestagingrole
+    type: gcp-service-account
+    explicit_name: gsa-staging
+    allowed_k8s_services:
+      - namespace: staging
+        service_account_name: default
+    links:
+      - gcs: [ "write" ]
+  - name: flytedevelopmentrole
+    type: gcp-service-account
+    explicit_name: gsa-development
+    allowed_k8s_services:
+      - namespace: development
+        service_account_name: default
+    links:
+      - gcs: [ "write" ]
+  - type: helm-chart
+    chart: "../../charts/flyte" # NOTE: relative path to chart
+    namespace: flyte
+    timeout: 600
+    create_namespace: true
+    values_file: "../../charts/flyte/values-gcp.yaml" # NOTE: relative path to values yaml
+    # Additional overrides to the values provided by the chart. Opta enables piping through produced outputs from prior modules/steps.
+    values:
+      postgres:
+        enabled: false
+      db:
+        datacatalog:
+          database:
+            port: 5432
+            username: "${{module.postgres.db_user}}"
+            host: "${{module.postgres.db_host}}"
+            dbname: "${{module.postgres.db_name}}"
+        admin:
+          database:
+            port: 5432
+            username: "${{module.postgres.db_user}}"
+            host: "${{module.postgres.db_host}}"
+            dbname: "${{module.postgres.db_name}}"
+      common:
+        ingress:
+          albSSLRedirect: false
+          host: "{parent.domain}"
+          annotations:
+            kubernetes.io/ingress.class: "nginx"
+            nginx.ingress.kubernetes.io/app-root: /console
+        databaseSecret:
+          secretManifest:
+            stringData:
+              pass.txt: "${{module.postgres.db_password}}"
+      storage:
+        bucketName: "${{module.gcs.bucket_name}}"
+        gcs:
+          projectId: "{vars.google_project_id}"
+      flyteadmin:
```

Comment:
@pmahindrakar-oss, don't we need a separate service account for flytescheduler?

Reply:
yes it should be separate service account , its a bug .

```
templates/flytescheduler/deployment.yaml:      serviceAccountName: {{ template "flyteadmin.name" . }}
```
I can fix that separately. The opta yaml should also contain deployment info for flytescheduler though which seems to be missing here

Reply:
Not once was the flyte scheduler mentioned here: https://docs.flyte.org/en/latest/deployment/gcp/index.html, so I did not add anything for it.

Reply:
Are there some revised instructions you wish to give?

Reply:
Also, I thought that we removed the special scheduler setup because it was using the default one since we used SQS in AWS and there currently was no equivalent in gcp?

Reply:
@juandiegopalomino currently it's a miss since we don't have a separate service account for flyte scheduler  and it uses the admin one.(I will fix that separately)
But the values-gcp.yaml has the deployment Info for the flytescheduler.

Since previous scheduler was dependent on aws features of sqs and cloud watch rules, we have created a native one which can be run on any cloud provider .

The values-gcp enables the native scheduler .

Reply:
awesome, so all values-gcp data in inherited (except where I explicitly overwrite), so is there anything I need to change in this PR?

Reply:
Can you confirm that flytescheuler pod comes up successfully when running with opta.
I can add the fix post for scheduler service account after your changes

Reply:
I will test out today with scheduler service account on my gcp deployment and update on this thread

Reply:
https://github.com/flyteorg/flyte/pull/1834
Here are the changes 
https://flyte--1834.org.readthedocs.build/en/1834/deployment/gcp/index.html#permissions

Reply:
The flyte scheduler comes up successfully. I don't see why this PR should be blocked by the (benign) bug in master or its bugfix-- it's working fine as it is.